build-system/server/app-utils.js

```diff
@@ -80,64 +82,81 @@ const isRtvMode = (serveMode) => {
-    // `.mjs` to be lazily built regardless of `.mjs`
-    file = file.replace(
-      /https:\/\/cdn\.ampproject\.org\/v0\.(m?js)/g,
-      hostName + '/dist/amp.js'
-    );
-    file = file.replace(
-      /https:\/\/cdn\.ampproject\.org\/shadow-v0\.(m?js)/g,
-      hostName + '/dist/amp-shadow.js'
-    );
-    file = file.replace(
-      /https:\/\/cdn\.ampproject\.org\/amp4ads-v0\.(m?js)/g,
-      hostName + '/dist/amp-inabox.js'
-    );
-    file = file.replace(
-      /https:\/\/cdn\.ampproject\.org\/video-iframe-integration-v0\.(m?js)/g,
-      hostName + '/dist/video-iframe-integration.js'
-    );
+function replaceUrls(mode, file, hostName, inabox) {
+  // If you need to add URL mapping logic, please don't do it in this function.
+  // Instead, do so in the `cdn.ts` module built below.
+
+  const cdnUrl = requireTsSync(
+    'build-system/server/new-server/transforms/utilities/cdn.ts'
+  );
+
+  // All inabox documents use `amp4ads-v0.js`. Its URL may be rewritten later.
+  if (inabox) {
     file = file.replace(
-      /https:\/\/cdn\.ampproject\.org\/v0\/(.+?)\.(m?js)/g,
-      hostName + '/dist/v0/$1.max.js'
+      /(https:\/\/cdn\.ampproject\.org)\/v0\.(m?js)/g,
+      '$1/amp4ads-v0.$2'
     );
-    if (inabox) {
-      const filename = '/dist/amp-inabox.js';
-      file = file.replace(/<html [^>]*>/, '<html amp4ads>');
-      file = file.replace(/\/dist\/amp\.(m?js)/g, filename);
+  }
+
+  hostName = hostName || '';
+
+  const pathnames = undefined; // we don't override the mapping, optional arg
+  const isRtv = isRtvMode(mode);
+
+  file = file.replace(
+    /https:\/\/cdn\.ampproject\.org\/.+\.(m?js)/g,
+    (match, extension) => {
+      // TODO(alanorozco): Match --esm in output extension and/or allow
+      // `.mjs` to be lazily built regardless of --esm
+      const url = new URL(match);
+      if (isRtv) {
+        return cdnUrl.CDNURLToRTVURL(url, mode, pathnames, extension);
+      }
+      const useMaxNames = mode !== 'compiled';
+      const {host, href, protocol} = cdnUrl.replaceCDNURLPath(
+        url,
+        pathnames,
+        extension,
+        useMaxNames
+      );
+      return hostName + href.substr(`${protocol}//${host}`.length);
     }
-  } else if (mode == 'compiled') {
-    file = file.replace(
-      /https:\/\/cdn\.ampproject\.org\/v0\.(m?js)/g,
-      hostName + '/dist/v0.$1'
-    );
-    file = file.replace(
-      /https:\/\/cdn\.ampproject\.org\/shadow-v0\.(m?js)/g,
-      hostName + '/dist/shadow-v0.$1'
-    );
-    file = file.replace(
-      /https:\/\/cdn\.ampproject\.org\/amp4ads-v0\.(m?js)/g,
-      hostName + '/dist/amp4ads-v0.$1'
-    );
-    file = file.replace(
-      /https:\/\/cdn\.ampproject\.org\/video-iframe-integration-v0\.(m?js)/g,
-      hostName + '/dist/video-iframe-integration-v0.$1'
-    );
-    file = file.replace(
-      /https:\/\/cdn\.ampproject\.org\/v0\/(.+?)\.(m?js)/g,
-      hostName + '/dist/v0/$1.$2'
-    );
+  );
+
+  // TODO(alanorozco): is this still useful?
```

Comment:
I believe this is used when you run `amp --compiled` (minified lazy-builder + dev server) and `amp server --compiled` (minified dev server). Might need to check if this is still true.

Reply:
Sure. The comment refers to whether the mappings under the condition are needed. @erwinmombay may have more context.

Reply:
Update:

1. The rule to remove `.max.js` can be excluded by changing the examples files to a replaceable CDN URL.
2. The rule to handle `.max.html` is of a kind that is not present in `new-server`, and should be ported forward. This PR does not change this since it's a new category of transform from `new-server`'s perspective.